### ISSUE TITLE
flocker with openstack cinder failed

### ISSUE THREAD

**Issue by fogray** [ScatterHQ/flocker#11]
Hi
 My docker is working on swarm mode. But when I using flocker volume plugin, run 'docker run -v vol1:/usr/local/tomcat/logs tomcat' command, it print error: 'docker: Error response from daemon: Timed out waiting for dataset to mount..'. 
My flocker-dataset-agent.service is running with error:
`Mar 27 03:33:51 flkn1 flocker-dataset-agent[541]: {"exception": "flocker.node.agents.blockdevice.UnknownInstanceID", "task_level": [2, 2, 5], "action_type": "flocker:agent:discovery", "reason": "Could not find valid instance ID for <flocker.node.agents.cinder.CinderBlockDeviceAPI object at 0x49d9c90>", "timestamp": 1490585631.412446, "task_uuid": "75a7ba2e-496f-43e1-a407-c31c8c6d249a", "action_status": "failed"}
Mar 27 03:33:51 flkn1 flocker-dataset-agent[541]: {"exception": "flocker.node.agents.blockdevice.UnknownInstanceID", "reason": "Could not find valid instance ID for <flocker.node.agents.cinder.CinderBlockDeviceAPI object at 0x49d9c90>", "timestamp": 1490585631.412774, "traceback": "Traceback: <class 'flocker.node.agents.blockdevice.UnknownInstanceID'>: Could not find valid instance ID for <flocker.node.agents.cinder.CinderBlockDeviceAPI object at 0x49d9c90>\n/opt/flocker/lib/python2.7/site-packages/machinist/_logging.py:102:receive\n/opt/flocker/lib/python2.7/site-packages/machinist/_fsm.py:534:receive\n/opt/flocker/lib/python2.7/site-packages/machinist/_fsm.py:592:output\n/opt/flocker/lib/python2.7/site-packages/flocker/node/_loop.py:529:output_CONVERGE\n--- <exception caught here> ---\n/opt/flocker/lib/python2.7/site-packages/twisted/internet/defer.py:150:maybeDeferred\n/opt/flocker/lib/python2.7/site-packages/flocker/node/agents/blockdevice.py:1798:discover_state\n/opt/flocker/lib/python2.7/site-packages/flocker/node/agents/blockdevice.py:1345:_count_calls\n/opt/flocker/lib/python2.7/site-packages/flocker/node/agents/blockdevice.py:1743:_discover_raw_state\n/opt/flocker/lib/python2.7/site-packages/flocker/node/agents/blockdevice.py:2026:compute_instance_id\n/opt/flocker/lib/python2.7/site-packages/flocker/node/agents/cinder.py:491:compute_instance_id\n", "message_type": "eliot:traceback", "task_uuid": "75a7ba2e-496f-43e1-a407-c31c8c6d249a", "task_level": [2, 3]}`

**Comment by wallnerryan** [comment]
are you running docker swarm from the nova compute nodes?

**Comment by fogray** [comment]
Yes, the nova compute node is running on a swarm work node. I have 3 nodes: node1, node2 and node3.
node1 is swarm manager node, node3 is openstack node running openstack components. And the nodes are also flocker nodes which node1 is also a flocker control node
